### ISSUE TITLE
Add default UI sound to NumberPicker

### ISSUE THREAD

**Issue by zepumph** [phetsims/scenery-phet#612]
This is needed for Ratio and Proportion sound design, see https://github.com/phetsims/ratio-and-proportion/issues/9#issuecomment-655167724. Tagging @jbphet 

Over in that issue, we decided that we will just use the basic and soft "bump" sound which is currently being used for when a slider moves across a tick line in Waves Intro.

**Comment by zepumph** [comment]
I just found that in Ratio and Proportion, I was just pulling the sound directly from waves-interference, most definitely meant to be a temporary measure. First I will factor out this sound into tambo, to be used by waves sims, RaP, and soon-to-be NumberPicker.

**Comment by zepumph** [comment]
In the above commits, I factored these sounds out to tambo. I kept the same names for clarity (especially if they still need to be tweaked). I think it is even more confusing because these sounds are called "clicks" but there are other "click" sounds already in tambo. We should make sure that the final names for these are clear to their purpose (not that we are yet complete sure what the purpose for them will be).

**Comment by zepumph** [comment]
After doing a bit more poking around I think it will be best to wait and pair on some of this with @jbphet. Here are some questions I want to discuss:
* Ensure I was correct in moving the Waves slider sounds into tambo (for use by NumberPicker and RaP)
* Take a look at the pattern of making a shared sound player for these sounds to be used by NumberPicker
* Should I also implement this in NumberSpinner?
* Note that there are already "click" sounds (and a shared sound player) in tambo, I would like to not step on "click" toes, perhaps that fills another role?
* Why set the default option to `null` and then sprinkle in the default sound player later?

Here is a patch I got working for NumberPicker. Hopefully we can take it and refine together when we meet Monday:

```diff
Index: scenery-phet/js/NumberPicker.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- scenery-phet/js/NumberPicker.js	(revision 4607ed50beb843e3dbfbc219d73429ace9a35cb2)
+++ scenery-phet/js/NumberPicker.js	(date 1594413313480)
@@ -29,6 +29,9 @@
 import PaintColorProperty from '../../scenery/js/util/PaintColorProperty.js';
 import AccessibleNumberSpinner from '../../sun/js/accessibility/AccessibleNumberSpinner.js';
 import SunConstants from '../../sun/js/SunConstants.js';
+import SharedSoundClip from '../../tambo/js/sound-generators/SharedSoundClip.js';
+import sliderClickSound from '../../tambo/sounds/slider-clicks-idea-c-example_mp3.js';
+import sliderBoundaryClickSound from '../../tambo/sounds/slider-clicks-idea-c-lower-end-click_mp3.js';
 import PhetioObject from '../../tandem/js/PhetioObject.js';
 import Tandem from '../../tandem/js/Tandem.js';
 import FireOnHoldInputListener from './buttons/FireOnHoldInputListener.js';
@@ -36,6 +39,10 @@
 import PhetFont from './PhetFont.js';
 import sceneryPhet from './sceneryPhet.js';
 
+const defaultSoundClipOptions = { initialOutputLevel: .2 }; // TODO: same as in waves-intro usage
+const sliderClickSoundClip = new SharedSoundClip( sliderClickSound, { soundClipOptions: defaultSoundClipOptions } );
+const sliderBoundaryClickSharedSoundClip = new SharedSoundClip( sliderBoundaryClickSound, { soundClipOptions: defaultSoundClipOptions } );
+
 /**
  * @param {Property.<number>} valueProperty
  * @param {Property.<Range>} rangeProperty - If the range is anticipated to change, it's best to have the range
@@ -116,6 +123,11 @@
     // Opacity used to indicate disabled, [0,1] exclusive
     disabledOpacity: SunConstants.DISABLED_OPACITY,
 
+    // {Playable|null} - Sound generators for when the NumberPicker's value changes, and when it hits range extremeties.
+    // If set to `null` the default sound will be used, use Playable.NO_SOUND to disable.
+    valueChangedSoundPlayer: sliderClickSoundClip,
+    minMaxSoundPlayer: sliderBoundaryClickSharedSoundClip,
+
     // phet-io
     tandem: Tandem.REQUIRED,
     phetioReadOnly: PhetioObject.DEFAULT_OPTIONS.phetioReadOnly,
@@ -319,6 +331,9 @@
   this.upListener = new FireOnHoldInputListener( {
     listener: function() {
       valueProperty.set( Math.min( options.upFunction( valueProperty.get() ), rangeProperty.get().max ) );
+
+      const soundClip = valueProperty.value === rangeProperty.get().max ? options.minMaxSoundPlayer : options.valueChangedSoundPlayer;
+      soundClip.play();
     },
     timerDelay: options.timerDelay,
     timerInterval: options.timerInterval
@@ -331,6 +346,9 @@
   this.downListener = new FireOnHoldInputListener( {
     listener: function() {
       valueProperty.set( Math.max( options.downFunction( valueProperty.get() ), rangeProperty.get().min ) );
+
+      const soundClip = valueProperty.value === rangeProperty.get().min ? options.minMaxSoundPlayer : options.valueChangedSoundPlayer;
+      soundClip.play();
     },
     timerDelay: options.timerDelay,
     timerInterval: options.timerInterval
Index: tambo/js/sound-generators/SharedSoundClip.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- tambo/js/sound-generators/SharedSoundClip.js	(revision 5cce2ac163b8c2ac4690183de0fedaeb0cacd554)
+++ tambo/js/sound-generators/SharedSoundClip.js	(date 1594412416768)
@@ -11,6 +11,7 @@
  * @author John Blanco (PhET Interactive Simulations)
  */
 
+import merge from '../../../phet-core/js/merge.js';
 import soundManager from '../soundManager.js';
 import tambo from '../tambo.js';
 import SoundClip from './SoundClip.js';
@@ -24,6 +25,11 @@
    */
   constructor( wrappedAudioBuffer, options ) {
 
+    options = merge( {
+      soundClipOptions: null,
+      soundManagerOptiosn: null
+    }, options );
+
     // {SoundClip} @private
     this.soundClip = new SoundClip( wrappedAudioBuffer, options.soundClipOptions );

**Comment by zepumph** [comment]
From meeting with @jbphet:
- [ ] The `.wav` files go into the `assets` folder, because they are easier to revamp/sample etc later (basically like including the reaper file but without breaking licensing terms of service).
- [ ] Rename these to "general-soft-click" and "general-bounds-boop"

**Comment by zepumph** [comment]
I implemented sounds in NumberPicker. One design question:

* Is it is OK to have the boundary sound repeat with keyboard nav if you hold down an arrow key and are at the max. This is pretty much a byproduct of how we tie sounds to UI interaction instead of the model Property changing. I don't think this should be changed, but would like a designer to sign off.

**Comment by zepumph** [comment]
I was looking for the original `wav` files for these sounds in wave-interference, and found https://github.com/phetsims/wave-interference/issues/484 as the issue they were added in, but didn't see any wav files associated with them. @Ashton-Morris could you please provide `wav` files for the following two sounds used for slider UI in wave-intereference:

* slider-clicks-idea-c-example.mp3
* slider-clicks-idea-c-lower-end-click.mp3

Also please note that these are moving to tambo (common code) to be used in other sims, and will be renamed to: 
* slider-clicks-idea-c-example.mp3 -> general-soft-click.mp3
* slider-clicks-idea-c-lower-end-click.mp3 -> general-boundary-boop.mp3

@jbphet mentioned that it is good to let you know about renames so that you can keep your reaper files aligned.

**Comment by zepumph** [comment]
I refactored WaveInterferenceSlider to use the general shared sound players. I decided not to do the same for RatioHalf.js and its usages of these sounds because those SoundClips are passed extra options like different enabledProperties and a different initialOutputLevel. @jbphet mentioned that changing these in one usage of a SharedSoundClip would change all usages, and I didn't want that leaking into the UI sound for the NumberPickers.

I think that all that is left for this issue is;
- [ ] @Ashton-Morris to give me `wav` files to add to the `assets/` in tambo for these two. Also for him to note the renaming.
- [ ] Design question in https://github.com/phetsims/scenery-phet/issues/612#issuecomment-657710112
- [ ] Get feedback about this UI component sound.
- [ ] @jbphet to do a code review of the above changes

**Comment by Ashton-Morris** [comment]
I have some .wav files 

[general-soft-click.wav](https://drive.google.com/file/d/1VI4FBZAgxfkVp5Nf8qBzUZ_RNyQljADS/view?usp=sharing)
[general-boundary-boop.wav](https://drive.google.com/file/d/1TyuS0aEoazu62YmTeH31THNGRyrzV-Xl/view?usp=sharing)

**Comment by zepumph** [comment]
>Is it is OK to have the boundary sound repeat with keyboard nav if you hold down an arrow key and are at the max. This is pretty much a byproduct of how we tie sounds to UI interaction instead of the model Property changing. I don't think this should be changed, but would like a designer to sign off.

At design meeting today, we decided that this behavior is ok, but we also recognized that there is some precedent from GFL mass sliders to not repeat the boundary sound. In that sim, those sounds are important to the concepts in the sim; they aren't just UI sound. With this in mind, we are planning to go ahead with allowing duplicate boundary sounds (this behavior is also in the slider implementation in Waves Intro). @Matthew-Moore240 @terracoda @BLFiedler please let us know if you want it to not repeat.


- [ ] This sound is being played on top GFL NumberPicker. We should fix that duplication.

**Comment by zepumph** [comment]
One more thing I forgot to mention in https://github.com/phetsims/scenery-phet/issues/612#issuecomment-657749792:

@Ashton-Morris, yesterday while working on this with @jbphet, we noticed that these two sounds are quite a bit quieter than @jbphet mentioned your standard is (-3dbfs). More around -5 or -6 dbfs, Do you think it would be a good idea to export these same sounds a bit louder?

**Comment by Ashton-Morris** [comment]
Ok I have exported them louder.

- [general-soft-click-001.wav](https://drive.google.com/file/d/1A2nj6ofrdCbsZBZJ02hQbHJL89hNtOph/view?usp=sharing)
- [general-boundary-boop-001.wav](https://drive.google.com/file/d/1H9xt3rldFwgwHCYK40hLlBftMMbHsypE/view?usp=sharing)

**Comment by zepumph** [comment]
Alright, I added the above to the assets of tambo, and I converted them to mp3 and replaced the sounds that were in sounds/ with those. I couldn't really hear a difference, so thanks for baring with me. It is good to be consistent though, I guess.

@jbphet can you please review my code changes to NumberPicker above, as well as the additions to tambo. 

* Should more files be in the assets directory? A brief glance at it for this issue seemed to show it not having nearly as many lossless files as it should. Should we create another issue to investigate having `wav` files for every production sound in `tambo/sounds/`?
* I'm most interested in you reviewing my strategy for implementing sound as a sound API user, please focus on  that.
* How is the naming of the sound?

Thanks!

**Comment by zepumph** [comment]
I'm unsure that this is complete, since discussing with @jbphet, I should be using the LAME encoding to convert `wav` files, and using 48 kbps. Please see and update `tambo/doc/encoding-audio.md` if needed.

**Comment by jbphet** [comment]
The code changes look reasonable.  It looks like there are several open questions and TODO items above, so I'll unassign myself and leave open.

**Comment by zepumph** [comment]
I created https://github.com/phetsims/tambo/issues/121, and I looked at the encoding I used to make the MP3 files, and it was using LAME.


I don't see anything else for this issue. Closing
